src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,238 @@
+  }
+
+  async createConversation(botId: string, userId: string, isCreatedByUser = false): Promise<any> {
+    const uid = Math.random()
+      .toString()
+      .substr(2, 6)
+    const title = `Conversation ${uid}`
+
+    await this.database.knex(conversationsTable).insert({
+      botId,
+      userId,
+      created_on: this.database.knex.date.now(),
+      last_heard_on: isCreatedByUser ? this.database.knex.date.now() : undefined,
+      title
+    })
+
+    const conversation = await this.database
+      .knex(conversationsTable)
+      .where({ title, userId, botId })
+      .select('id')
+      .get(0)
+
+    return conversation && conversation['id']
+  }
+
+  async appendUserMessage(
+    botId: string,
+    userId: string,
+    conversationId: string,
+    fullName: string,
+    avatarUrl: string,
+    payload: sdk.conversations.MessagePayload
+  ) {
+    const convo = await this.database
+      .knex(conversationsTable)
+      .where({ userId, id: conversationId, botId })
+      .select('id')
+      .limit(1)
+      .get(0)
+
+    if (!convo) {
+      throw new Error(`Conversation "${conversationId}" not found`)
+    }
+
+    const { type, text, raw, data } = payload
+
+    const message = {
+      id: uuid.v4(),
+      conversationId,
+      userId,
+      full_name: fullName,
+      avatar_url: avatarUrl,
+      message_type: type,
+      message_text: text,
+      message_raw: this.database.knex.json.set(raw),
+      message_data: this.database.knex.json.set(data),
+      sent_on: this.database.knex.date.now()
+    }
+
+    return Promise.join(
+      this.database.knex(messageTable).insert(message),
+      this.database
+        .knex(conversationsTable)
+        .where({ id: conversationId, userId: userId, botId: botId })
+        .update({ last_heard_on: this.database.knex.date.now() }),
+      () => ({
+        ...message,
+        sent_on: new Date(),
+        message_raw: raw,
+        message_data: data
+      })
+    )
+  }
+
+  async appendBotMessage(
+    botName: string,
+    botAvatar: string,
+    conversationId: string,
+    payload: sdk.conversations.MessagePayload
+  ): Promise<any> {
+    const { type, text, raw, data } = payload
+    const message = {
+      id: uuid.v4(),
+      conversationId: conversationId,
+      userId: undefined,
+      full_name: botName,
+      avatar_url: botAvatar,
+      message_type: type,
+      message_text: text,
+      message_raw: this.database.knex.json.set(raw),
+      message_data: this.database.knex.json.set(data),
+      sent_on: this.database.knex.date.now()
+    }
+
+    await this.database.knex(messageTable).insert(message)
+
+    return Object.assign(message, {
+      sent_on: new Date(),
+      message_raw: this.database.knex.json.get(message.message_raw),
+      message_data: this.database.knex.json.get(message.message_data)
+    })
+  }
+
+  async listConversations(userId: string, botId: string) {
```

Comment:
we might want to pass a limit param instead of hardcoding to 100

Reply:
i'd suggest a default param, because it won't be changed often